pkgs/by-name/ha/harlequin/package.nix

```diff
@@ -0,0 +1,96 @@
+{
+  lib,
+  python3,
+  fetchFromGitHub,
+  withPostgresAdapter ? true,
+  withBigQueryAdapter ? true,
+}:
+python3.pkgs.buildPythonApplication rec {
+  pname = "harlequin";
+  version = "1.24.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "tconbeer";
+    repo = "harlequin";
+    rev = "v${version}";
+    hash = "sha256-exgpdixYc0hpB0GGw20Apg1rlw3lhStJaIX0UHPu10A=";
+  };
+
+  build-system = with python3.pkgs; [
+    poetry-core
+  ];
+
+  dependencies =
+    with python3.pkgs;
+    [
+      textual
+      textual-fastdatatable
+      textual-textarea
+      click
+      rich-click
+      duckdb
+      sqlfmt
+      platformdirs
+      importlib-metadata
+      tomlkit
+      questionary
+      numpy
+      packaging
+    ]
+    ++ lib.optionals withPostgresAdapter [ harlequin-postgres ]
+    ++ lib.optionals withBigQueryAdapter [ harlequin-bigquery ];
+
+  passthru.optional-dependencies = with python3.pkgs; {
+    postgres = [
+      harlequin-postgres
+    ];
+    bigquery = [
+      harlequin-bigquery
+    ];
+    # adbc = [
+    #   harlequin-adbc
+    # ];
+    # cassandra = [
+    #   harlequin-cassandra
+    # ];
+    # databricks = [
+    #   harlequin-databricks
+    # ];
+    # mysql = [
+    #   harlequin-mysql
+    # ];
+    # nebulagraph = [
+    #   harlequin-nebulagraph
+    # ];
+    # odbc = [
+    #   harlequin-odbc
+    # ];
+    # s3 = [
+    #   boto3
+    # ];
+    # trino = [
+    #   harlequin-trino
+    # ];
+  };
+
+  pythonRelaxDeps = [
+    "textual"
+  ];
+
+  pythonImportsCheck = [
+    "harlequin"
+    "harlequin_duckdb"
+    "harlequin_sqlite"
+    "harlequin_vscode"
+  ];
+
+  meta = {
```

Comment:
You should add `mainProgram` to indicate the name of the main executable provided by this package.